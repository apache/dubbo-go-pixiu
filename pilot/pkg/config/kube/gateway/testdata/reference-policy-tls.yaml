apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: cross
    hostname: "cert.domain.example"
    port: 443
    protocol: HTTPS
    allowedRoutes:
      namespaces:
        from: Selector
        selector:
          matchLabels:
            kubernetes.io/metadata.name: "cert"
    tls:
      mode: Terminate
      certificateRefs:
      - name: cert
        namespace: cert
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: ReferencePolicy
metadata:
  name: alllow-cert
  namespace: cert
spec:
  from:
  - group: networking.gateway.k8s.io
    kind: Gateway
    namespace: istio-system
  to:
    - group: ""
      kind: Secret
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: cert
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  hostnames: ["cert.domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80
