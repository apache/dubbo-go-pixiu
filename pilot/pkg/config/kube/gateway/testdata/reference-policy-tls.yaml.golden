apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parent: Gateway/gateway/cross.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-cross
  namespace: istio-system
spec:
  servers:
  - hosts:
    - cert/cert.domain.example
    port:
      name: default
      number: 443
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://cert/cert
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/http.cert
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: cert
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-cross
  hosts:
  - cert.domain.example
  http:
  - route:
    - destination:
        host: httpbin.cert.svc.domain.suffix
        port:
          number: 80
---
